newIDE/app/src/SceneEditor/index.js

```diff
@@ -933,9 +962,17 @@ export default class SceneEditor extends Component {
           ref={contextMenu => (this.contextMenu = contextMenu)}
           buildMenuTemplate={() => [
             {
-              label: 'Edit Object',
+              label:
+                'Add an Instance of ' +
+                this.shortenedString(this.state.selectedObjectNames[0], 7),
+              click: () => this._onAddInstanceUnderCursor(),
+              visible: this.state.selectedObjectNames.length > 0,
+            },
+            {
+              label: 'Edit Object ' +
+              this.shortenedString(this.state.selectedObjectNames[0], 14),
               click: () => this.openObjectEditor(),
-              enabled: this.instancesSelection.hasSelectedInstances(),
+              visible: this.instancesSelection.hasSelectedInstances(),
```

Comment:
Are you absolutely 100% sure that when `openObjectEditor` is called, it will open the editor of `this.state.selectedObjectNames[0]`?

If I look at what `openObjectEditor` is doing, it's opening the editor of the object associated to the first instance in the list of selected instances. There is no guarantee that it will be the same as the first object name in the list of selected objects.

I think that instead you want to change click: () => this.openObjectEditor to call `() => this.editObjectByName(this.state.selectedObjectNames[0])` and change the visible condition to `this.state.selectedObjectNames.length > 0`?

Reply:
> Also, I have a question, unrelated to the code. As this behavior is now different than the previous version, existing users might be confused why they can't click on an object and then click on the scene to insert it like before.
> Both the drag'n'drop and the right click are not too difficult to discover but I still think we should add an extra "hint".
> 
> I'm thinking of this: when you click on the background with an object selected, there is a `Snackbar` showing in the bottom (like when you click on an object) telling that you can "Right click or drag'n'drop to insert an instance".
> 
> What do you think? :)

Thank you for reviewing this , will have a look after work :) 
We actually already have a hint. When an object is selected from the list, this message comes up:
https://github.com/4ian/GDevelop/pull/651/files#diff-f1390b754016cd00493b4bf4b2c08469R889
It is now actually much easier to insert instances, as the user has multiple different ways of doing it. I don't think they will be confused.

If you click in the background, it clears instance selection, but if a hint about this option being in the r.click menu constantly appears every time when that happens- It might become annoying to the user.
I think the option being in that menu is very easy to discover to begin with, but maybe we can think of something if the feedback is that it isn't

Reply:
This way the two menu functions follow are more similar pattern too. This makes the code easier to read

Reply:
Yes code correctness is important here, because things are evolving, so something that "always work now" will probably break later when someone (me, you or even other some developers) rework this and miss this details. (for example, if I refactor the selected instances, I could be changing/removing this function to check if instances are selected, and then because it's not properly linked to the selected object names anymore, boom when we click on the button there is no object selected and the code start using an empty array while it should not be empty - guaranteed to crash the editor at some point :))

Reply:
@4ian This is a very insightful observation and good design ethic. Thank you for sharing it.
You are absolutely right - the more organised the code, the easier to refactor and contribute to without introducing regressions  :) 
In that I am really happy when you review a pull and present a better approach to some things

Reply:
👍